### DISCUSSION TITLE
TypographyStylesProvider -- should this support `size` attribute?

### DISCUSSION THREAD

**Discussion by cstrat** [mantinedev/mantine#81] [Q&A] [answered]
Just wondering if `TypographyStylesProvider` should support the `size` attribute to be aligned with `<Text>` sizing?
Without this attribute it is going to be hard to have a consistent font-size, I could manually set it - but surely this is possible in the component?

````
<Text size="sm">This is the same size as below...</Text>
<TypographyStylesProvider size="sm">
  <ul>
    <li>Same size as above!</li>
  </ul>
</TypographyStylesProvider>
````

**Comment by rtivital** [comment, marked as answer]
This is a tricky part, it includes a bunch of other elements – headings, blockquotes, etc. which do not support size, not sure how it can be implemented well to keep a consistent rhythm

**Reply by cstrat** [reply]
So in the docs it says:

> TypographyStylesProvider includes styles for:
> paragraphs
> headings
> lists
> blockquotes
> tables
> links
> images
> hr
> code and pre

Paragraphs we have `<Text>` although perhaps a `<Paragraph>` would be nice?
Headings we have `<Title>`
Blockquotes we have `<Blockquote>`
Tables we have `<Table>`
Links we have `<Anchor>`
Images we have `<Image>`
Hr we have `<Hr>`
Code we have `<Code>`

The only thing we don't have a component for is lists.
If lists had a component wrapper (like Table) then I am not even sure what we would need `TypographyStylesProvider` for.

**Reply by cstrat** [reply]
If you were to add `<List>`. It would be good to support not only `<ul>` and `<ol>` but I often use `<dl>` too.
Perhaps have `<List component={<dl>}>` or something like that, similar to how `<Overlay>` lets you compose using different elements.

**Reply by rtivital** [reply]
The tricky part as that although we have all these components, most of them do not support sizes (Title, Blockquote, Table).
TypographyStylesProvider has one breakpoint now that shrinks font-size from md to sm, and I think it will be better to do something like `scale` instead of size:

```tsx
<TypographyStylesProvider scale="md" />; // default font-sizes
<TypographyStylesProvider scale="xl" />; // all font-sizes * 1.5 (predefined xl value)
<TypographyStylesProvider scale="sm" />; // all font-sizes * 0.9 (predefined sm value)
<TypographyStylesProvider scale={0.7} />; // all font-sizes * 0.7
```

**Reply by cstrat** [reply]
Yeah that would work.

Sorry I think I missed my point in the post above.
My point above was that you only really **must** use `TypographyStylesProvider` for lists as everything else has a native mantine component. Why would you use this component for anything else? I know there is the edge case where you are importing html from external and want it styled consistently.

**Reply by rtivital** [reply]
This component is designed to be used with html content that requires styling but does not have any associated classNames, e.g. in one of my projects I have a CMS where user can edit rich text and then this html is inserted to page with these styles without exposing them to global scope. 

About list – it'll be added in 1.2.0 (in July) so I guess it will solve your issue with TypographyStylesProvider

**Reply by cstrat** [reply]
I understand your scenario - sounds like this is a very elegant solution to that problem.

> About list – it'll be added in 1.2.0 (in July) so I guess it will solve your issue with TypographyStylesProvider

That's great, thanks, that will be my fix 😄
